rust/src/delta_datafusion.rs

```diff
@@ -373,25 +375,44 @@ impl TableProvider for DeltaTable {
             });
         };
 
-        let table_partition_cols = self.get_metadata()?.partition_columns.clone();
+        let table_partition_cols = self
+            .get_metadata()?
+            .partition_columns
+            .clone()
+            .into_iter()
+            .flat_map(|c| {
+                let f_types = schema
+                    .fields()
+                    .iter()
+                    .map(|e| (e.name(), e.data_type()))
+                    .filter(|&(col_name, _)| c.eq_ignore_ascii_case(col_name))
+                    .collect::<Vec<_>>();
+                f_types.first().map(|o| o.to_owned())
+            })
+            .collect::<HashMap<_, _>>();
         let file_schema = Arc::new(ArrowSchema::new(
             schema
                 .fields()
                 .iter()
-                .filter(|f| !table_partition_cols.contains(f.name()))
+                .filter(|f| !table_partition_cols.contains_key(f.name()))
                 .cloned()
                 .collect(),
         ));
-        let parquet_scan = ParquetFormat::default()
+
+        let parquet_scan = ParquetFormat::new(session.config_options())
             .create_physical_plan(
                 FileScanConfig {
                     object_store_url: self.storage.object_store_url(),
                     file_schema,
                     file_groups: file_groups.into_values().collect(),
                     statistics: self.datafusion_table_statistics(),
-                    projection: projection.clone(),
+                    projection: projection.map(|o| o.to_owned()),
```

Comment:
My suggestion here would be to just clone the partition cols, then use them to filter the file schema fields. and afterwards use them to create the tuple array needed. Thus we avoid creating a unnecessary intermediate structure.